doc/developer-guide.md

```diff
@@ -37,18 +38,45 @@ Axe 3.0 supports open Shadow DOM: see our virtual DOM APIs and test utilities fo
 
+## Watching for Changes
+
+You can watch for changes and automatically build axe and run relevant tests using `npm run develop`. Once run, any changes to files inside the [lib directory](../lib) will rebuild axe. After axe is built, it will try to run the relevant tests for the files changed. If you change a file inside the [test directory](../test) it will run the tests for the file changed.
+
+There is one exception to automatically run tests for changed files. Changes to files in the [full integration test directory](../test/integration/full) will not run the tests. This is because these tests require the browser to navigate to the page directly, which is something Mocha / Karma does not support.
+
+**Note:** We are still working on improving knowing which tests are relevant to the changed file so this may not correctly run tests every time. In these cases you should run the tests manually.
+
 ### Running Tests
 
-To run all tests from the command line you can run `npm test`, which will run all unit and integration tests using headless chrome and Selenium Webdriver.
+To run all tests from the command line you can run `npm test`, which will run all unit and integration tests using headless Chrome. Having axe built and up-to-date is required in order to run tests. If you update files inside the [lib directory](../lib) you will need to rebuild axe before running tests.
+
+You can scope which set of tests to run through various npm scripts: 
+
+- `npm run test:unit:core` - Run only [core tests](../test/core/)
+- `npm run test:unit:commons` - Run only [commons tests](../test/commons/)
+- `npm run test:unit:checks` - Run only [check tests](../test/checks/)
+- `npm run test:unit:rule-matches` - Run only [rule matches](../test/rule-matches/)
+- `npm run test:unit:integration` - Run only [integration rule tests](../test/integration/rules/)
+- `npm run test:unit:virtual-rules` - Run only [virtual rule tests](../test/integration/virtual-rules)
+- `npm run test:unit:api` - Run only [api tests](../test/integration/api)
+
+Additionally, you can [watch for changes](#watching-for-changes) to files and automatically run the relevant tests.
+
+If you need to debug a test in the browser, you can run `npm run test:debug` which will run the Karma tests in Chrome. You can scope which set of tests to run by passing the `testDirs` argument. Supported values are:
+
+- `core`
+- `commons`
+- `checks`
+- `rule-matches`
+- `integration`
+- `virtual-rules`
+- `api`
+
+Example:
 
-You can also load tests in any supported browser, which is helpful for debugging. Tests require a local server to run, you must first start a local server to serve files. You can use Grunt to start one by running `npm start`. Once your local server is running you can load the following pages in any browser to run tests:
+- `npm run test:debug -- testDirs=core`
```

Comment:
```suggestion
- `npm run test:debug -- testDirs=core`

You can then open the inspector and debug normally in Chrome by setting breakpoints and refreshing the page. Mocha will re-execute the tests with your debugger running.
```

Reply:
I feel like this part may not be necessary. Do you feel people debugging tests in a browser wouldn't know this and should to be informed about it?